sdk/scheduler/src/main/java/com/mesosphere/sdk/api/EndpointsResource.java

```diff
@@ -149,27 +150,37 @@ public Response getEndpoint(
             // TODO(mrb): Also extract DiscoveryInfo from executor, when executors get the ability to specify resources
             DiscoveryInfo discoveryInfo = taskInfo.getDiscovery();
             // Mesos DNS hostname:
-            String mesosDnsHost = String.format("%s.%s.mesos",
+            String mesosDnsHost = String.format("%s.%s.%s",
                     (taskInfo.hasDiscovery() && taskInfo.getDiscovery().hasName())
                     ? taskInfo.getDiscovery().getName()
                     : taskInfo.getName(),
-                    serviceName);
+                    serviceName,
+                    Constants.DNS_TLD);
             // Hostname of agent at offer time:
             String nativeHost = new SchedulerLabelReader(taskInfo).getHostname();
-
+            List<String> ipAddresses = new ArrayList<>();
+            Protos.TaskStatus taskStatus = stateStore.fetchStatus(taskInfo.getName()).orElse(null);
```

Comment:
I thought we were going to store IPs / TaskStatuses with IP addresses elsewhere.